### PR TITLE
can skip validating a bookmark

### PR DESCRIPTION
**Pull request by superyyrrzz**

### CODE REVIEW

File: src/Microsoft.DocAsCode.Build.Engine/PostProcessors/ValidateBookmark.cs [hunk 1]
```diff
@@ -78,6 +79,8 @@ protected override void HandleCore(HtmlDocument document, ManifestItem manifestI
             // RFC 3986: relative-ref = relative-part [ "?" query ] [ "#" fragment ]
             _linksWithBookmark[outputFile] =
                 (from node in GetNodesWithAttribute(document, "href")
+                 let nocheck = node.GetAttributeValue("nocheck", null)
+                 where !"bookmark".Equals(nocheck, StringComparison.OrdinalIgnoreCase)
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.Build.Engine/PostProcessors/ValidateBookmark.cs]: "bookmark" to const

File: test/Microsoft.DocAsCode.Build.Engine.Tests/ValidateBookmarkTest.cs [hunk 2]
```diff
@@ -79,6 +79,45 @@ public void TestBasicFeature()
...
+                new HtmlPostProcessor
+                {
+                    Handlers = { new ValidateBookmark() }
+                }.Process(manifest, _outputFolder);
+            }
+            Logger.UnregisterListener(_listener);
+
+            // Assert
+            var logs = _listener.Items;
+            Console.WriteLine(string.Concat(logs.Select(l => Tuple.Create(l.Message, l.File))));
```
Comment (hellosnow) [inline test/Microsoft.DocAsCode.Build.Engine.Tests/ValidateBookmarkTest.cs]: >Console [](start = 12, length = 7)

why console.writeline?

Comment (ansyral) [inline test/Microsoft.DocAsCode.Build.Engine.Tests/ValidateBookmarkTest.cs]: was for debugging before

**Comment by ansyral** [comment]
:shipit:

**Comment by hellosnow** [comment]
:shipit:

**Comment by vicancy** [comment]
>                 foreach (var attr in node.ChildAttributes("sourceFile"))

looks like the debug attributes can be gathered in an array

---
Refers to: src/Microsoft.DocAsCode.Build.Engine/PostProcessors/RemoveDebugInfo.cs:20 in 40830dd. [](commit_id = 40830ddb395d7915f99adb1c135ef2c1ae66a7cc, deletion_comment = False)

File: test/Microsoft.DocAsCode.Build.Engine.Tests/ValidateBookmarkTest.cs [hunk 3]
```diff
@@ -83,6 +82,44 @@ public void TestBasicFeature()
...
+            // Act
+            Logger.RegisterListener(_listener);
+            using (new LoggerPhaseScope("validate_bookmark"))
+            {
+                new HtmlPostProcessor
+                {
+                    Handlers = { new ValidateBookmark() }
+                }.Process(manifest, _outputFolder);
+            }
+            Logger.UnregisterListener(_listener);
```
Comment (vicancy) [inline test/Microsoft.DocAsCode.Build.Engine.Tests/ValidateBookmarkTest.cs]: >UnregisterListener [](start = 19, length = 18)

move to finally?

File: src/Microsoft.DocAsCode.Build.Engine/PostProcessors/RemoveDebugInfo.cs [hunk 4]
```diff
@@ -9,6 +9,14 @@ namespace Microsoft.DocAsCode.Build.Engine
 
     public sealed class RemoveDebugInfo : HtmlDocumentHandler
     {
+        private readonly string[] RemoveAttributes =
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.Build.Engine/PostProcessors/RemoveDebugInfo.cs]: naming: DebugInfoAttributes?

**Pull request merged**